airflow/configuration.py

```diff
@@ -530,6 +530,25 @@ def parameterized_config(template):
 
 conf.read(AIRFLOW_CONFIG)
 
+if 'AIRFLOW_HOME' in os.environ:
+    if conf.has_option('core', 'AIRFLOW_HOME'):
```

Comment:
Since you check this within both the if and the else, you could place it at the top, and check for `AIRFLOW_HOME` env var inside, to get rid of 1 if statement:
```
if conf.has_option('core', 'AIRFLOW_HOME'):
    if 'AIRFLOW_HOME' in os.environ:
        ...
    else:
        ...
```